setting Python3_LIBRARY is broken in virtual environments in MSYS2

I tried building a project with scikit-build 0.17.1 using `python -m build` which builds in a virtual environment in mingw on windows. It fails while configuring cmake. Here is the relevant output.

```
Configuring Project
  Working directory:
    C:/msys64/tmp/build-via-sdist-oo6vlkz6/winsdk-1.0.0b8/_skbuild/mingw_aarch64-3.10/cmake-build
  Command:
    C:/msys64/tmp/build-env-cx3hyuyr/lib/python3.10/site-packages/cmake/data/bin/cmake.exe C:/msys64/tmp/build-via-sdist-oo6vlkz6/winsdk-1.0.0b8 -G Ninja --no-warn-unused-cli -DCMAKE_INSTALL_PREFIX:PATH=C:/msys64/tmp/build-via-sdist-oo6vlkz6/winsdk-1.0.0b8/_skbuild/mingw_aarch64-3.10/cmake-install -DPYTHON_VERSION_STRING:STRING=3.10.11 -DSKBUILD:INTERNAL=TRUE -DCMAKE_MODULE_PATH:PATH=C:/msys64/tmp/build-env-cx3hyuyr/lib/python3.10/site-packages/skbuild/resources/cmake -DPYTHON_EXECUTABLE:PATH=C:/msys64/tmp/build-env-cx3hyuyr/bin/python.exe -DPYTHON_INCLUDE_DIR:PATH=C:/msys64/clangarm64/include/python3.10 -DPYTHON_LIBRARY:PATH=libpython3.10.a -DPython_EXECUTABLE:PATH=C:/msys64/tmp/build-env-cx3hyuyr/bin/python.exe -DPython_ROOT_DIR:PATH=C:/msys64/tmp/build-env-cx3hyuyr -DPython_FIND_REGISTRY:STRING=NEVER -DPython_INCLUDE_DIR:PATH=C:/msys64/clangarm64/include/python3.10 -DPython_LIBRARY:PATH=libpython3.10.a -DPython3_EXECUTABLE:PATH=C:/msys64/tmp/build-env-cx3hyuyr/bin/python.exe -DPython3_ROOT_DIR:PATH=C:/msys64/tmp/build-env-cx3hyuyr -DPython3_FIND_REGISTRY:STRING=NEVER -DPython3_INCLUDE_DIR:PATH=C:/msys64/clangarm64/include/python3.10 -DPython3_LIBRARY:PATH=libpython3.10.a -DCMAKE_MAKE_PROGRAM:FILEPATH=C:/msys64/tmp/build-env-cx3hyuyr/lib/python3.10/site-packages/ninja/data/bin/ninja -DCMAKE_BUILD_TYPE:STRING=Release

Not searching for unused variables given on the command line.
-- The CXX compiler identification is MSVC 19.35.32217.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.35.32215/bin/Hostx86/arm64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at C:/msys64/tmp/build-env-cx3hyuyr/lib/python3.10/site-packages/cmake/data/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Python3 (missing: Python3_LIBRARIES Development.Module)
  (found version "3.10.11")
Call Stack (most recent call first):
  C:/msys64/tmp/build-env-cx3hyuyr/lib/python3.10/site-packages/cmake/data/share/cmake-3.26/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  C:/msys64/tmp/build-env-cx3hyuyr/lib/python3.10/site-packages/cmake/data/share/cmake-3.26/Modules/FindPython/Support.cmake:3766 (find_package_handle_standard_args)
  C:/msys64/tmp/build-env-cx3hyuyr/lib/python3.10/site-packages/cmake/data/share/cmake-3.26/Modules/FindPython3.cmake:551 (include)
  CMakeLists.txt:32 (find_package)


-- Configuring incomplete, errors occurred!
```

The error `Could NOT find Python3 (missing: Python3_LIBRARIES Development.Module)` is caused by `-DPython3_LIBRARY:PATH=libpython3.10.a` which is not a full path.

I did some debugging and it looks like the problem is related to this old TODO: https://github.com/scikit-build/scikit-build/blob/84c7e4c7817034e39f5a6b6d86d3f2e9c3ec926b/skbuild/cmaker.py#L605

Most code paths in `CMaker.get_python_library()` check that the library path actually exists. But since the library is not found in a virtual environment, we end up with the default value from `_guess_python_library()` falling through. 

To reproduce, run the following in a virtualenv.

```python
from skbuild.cmaker import CMaker

print(CMaker.get_python_library("3.10"))
```

There is some relevant discussion about why virtualenvs don't have a copy of the library at https://github.com/python/cpython/issues/87500

So, it seems like in a virtualenv, `get_python_library()` should always return `None`.

--- a/skbuild/cmaker.py
+++ b/skbuild/cmaker.py
@@ -297,7 +297,7 @@ class CMaker:
             )
             if python_include_dir:
                 cmd.append(f"{prefix}_INCLUDE_DIR:PATH={python_include_dir}")
-            if python_library and sys.platform.startswith("win"):
+            if python_library and sysconfig.get_platform().startswith("win"):
                 cmd.append(f"{prefix}_LIBRARY:PATH={python_library}")
             if sys.implementation.name == "pypy":
                 cmd.append(f"{prefix}_FIND_IMPLEMENTATIONS:STRING=PyPy")

--- a/skbuild/constants.py
+++ b/skbuild/constants.py
@@ -43,7 +43,7 @@ def _default_skbuild_plat_name() -> str:
 
     On macOS, it corresponds to the version and machine associated with :func:`platform.mac_ver()`.
     """
-    if sys.platform != "darwin":
+    if not sys.platform.startswith("darwin"):
         return get_platform()
 
     supported_macos_architectures = {"x86_64", "arm64"}

